opensearch-operator/pkg/helpers/reconcile-helpers.go

```diff
@@ -40,35 +40,15 @@ func ResolveImage(cr *opsterv1.OpenSearchCluster, nodePool *opsterv1.NodePool) (
 	defaultRepo := "docker.io/opensearchproject"
 	defaultImage := "opensearch"
 
-	var version string
-
 	// If a general custom image is specified, use it.
 	if cr.Spec.General.ImageSpec != nil {
 		if useCustomImage(cr.Spec.General.ImageSpec, &result) {
 			return
 		}
 	}
 
-	// Calculate version based on upgrading status
```

Comment:
why we deleted that func? dont we need it ?

Reply:
With the new way the upgrade is done it's no longer needed to set the version based on upgrade status. Instead all STS are updated immediately and the operator only restarts the pods one-by-one to apply the changes.